src/backend/access/zedstore/zedstore_attstream.c

```diff
@@ -166,13 +166,15 @@ decode_attstream_begin(attstream_decoder *decoder, ZSAttStream *attstream)
 					  attstream->t_size - SizeOfZSAttStreamHeader,
 					  attstream->t_decompressed_bufsize);
 		decoder->chunks_len = attstream->t_decompressed_size;
+		decoder->compression_ratio = ((float8) buf_size_needed) / attstream->t_size;
```

Comment:
The compression ratio here is only the ratio for zedstore's compression, in-line toasted compression is not considered here, which is probably fine for our purposes. I'm not sure about out-of-line toast pages though, do we need to included the cost of reading the toast page that a toast datum points to?

Reply:
Yes, that's a problem, we have extra blocks in zedstore files include TID blocks, toast blocks and redo blocks, we need a way to consider those blocks. Do heap table considered the cost of toast blocks?